app/controllers/needs_controller.rb

```diff
@@ -20,8 +20,8 @@ def index
     else
       @needs = @needs.order(created_at: :desc)
     end
-
-    @needs = @needs.page(params[:page])
+    page_size = params[:page_size].to_i > 0 ? params[:page_size] : 10
```

Comment:
1. We'll want to add `:page_size` to `params.permit` above to avoid console errors
2. This sets `page_size` as a string if one is provided, and a number if one isn't
3. This allows page sizes of any number (e.g. 99999999999) rather than restricting it to the 4 options in your helper